src/cmake.ts

```diff
@@ -317,24 +358,14 @@ export class CMakeTools {
     /**
      * @brief Initializes the 'selectedBuildType' attribute.
      *
-     * The 'selectedBuildType' attribute is a bit of a tricky thing. On initial
-     * load, we want to read it from the CMake cache, unless we are in a
-     * multiconf generator like Visual Studio. In that case, we will default to
-     * 'Debug'.
-     *
-     * This should only be called when the build type is completely impossible
-     * to determine without looking into the CMake cache.
-     *
      * @returns A promise resolving to the selectedBuildType
      */
     public _initSelectedBuildType = async function (): Promise<string> {
         const self: CMakeTools = this;
-        if (!(await self.cache.exists()))
-            self.selectedBuildType = 'None';
-        if (await self.isMultiConf())
-            self.selectedBuildType = 'Debug';
-        else
-            self.selectedBuildType = (await self.cache.get('CMAKE_BUILD_TYPE')).as<string>();
+        if (await self.cache.exists() && await self.isMultiConf())
+            if (self.selectedBuildType === 'None')
+                self.selectedBuildType = 'Debug';
+
         return self.selectedBuildType;
```

Comment:
Yes I agree. In fact `.cmaketools.json` is already checked for in the constructor and defaults `selectedBuildType` to `None` (check the line where _`extCache` is constructed).
I pushed ca0cec417e5dd62a021434c66a7d6a09ae18e65e to load `selectedBuildType` from `CMakeCache.txt` if it is `None`